src/server/bitops_family.cc

```diff
@@ -0,0 +1,267 @@
+#include "server/bitops_family.h"
+
+#include <bitset>
+
+#include "server/string_family.h"
+extern "C" {
+#include "redis/object.h"
+}
+
+#include "base/logging.h"
+#include "server/command_registry.h"
+#include "server/common.h"
+#include "server/conn_context.h"
+#include "server/engine_shard_set.h"
+#include "server/error.h"
+#include "server/tiered_storage.h"
+#include "server/transaction.h"
+#include "util/varz.h"
+
+namespace dfly {
+using namespace facade;
+
+namespace {
+static const int32_t OFFSET_FACTOR = 8;  // number of bits in byte
+static const long INDEX_MAPPING[] = {    // or same reason at REDIS the indenes at left to right
+    7, 6, 5, 4, 3, 2, 1, 0};
+// the offset is in bites
+// The following is the list of the functions that would handle the
+// commands that handle the bit operations
+void BitPos(CmdArgList args, ConnectionContext* cntx);
+void BitCount(CmdArgList args, ConnectionContext* cntx);
+void BitField(CmdArgList args, ConnectionContext* cntx);
+void BitFieldRo(CmdArgList args, ConnectionContext* cntx);
+void BitOp(CmdArgList args, ConnectionContext* cntx);
+void GetBit(CmdArgList args, ConnectionContext* cntx);
+void SetBit(CmdArgList args, ConnectionContext* cntx);
+OpResult<std::string> ReadValue(const OpArgs& op_args, std::string_view key);
+std::string GetString(EngineShard* shard, const PrimeValue& pv);
+// ------------------------------------------------------------------------- //
+//  Bits manipulation functions
+constexpr int32_t GetBitIndex(int32_t offset) noexcept {
+  return offset % OFFSET_FACTOR;
+}
+
+constexpr int32_t GetByteIndex(int32_t offset) noexcept {
+  return offset / OFFSET_FACTOR;
+}
+
+std::string::value_type GetByteValue(const std::string& value, int32_t offset) {
+  return value[GetByteIndex(offset)];
+}
+
+long GetBitValue(const std::string& value, int32_t offset) {
+  const std::bitset<8> bits{static_cast<unsigned long long>(GetByteValue(value, offset))};
+  return static_cast<long>(bits.test(INDEX_MAPPING[GetBitIndex(offset)]));
+}
+
+long GetBitValueSafe(const std::string& value, int32_t offset) {
+  return ((value.size() * OFFSET_FACTOR) > (std::size_t)offset) ? GetBitValue(value, offset) : 0;
+}
+
+long SetBitValue(std::string& value, int32_t offset, int32_t new_value) {
+  // we need to return the old value after setting the value for offset
+  const auto old_value{GetBitValue(value, offset)};  // save this as the return value
+  std::bitset<8> bits{static_cast<unsigned long long>(GetByteValue(value, offset))};
+  bits.set(INDEX_MAPPING[GetBitIndex(offset)], new_value != 0);
+  value[GetByteIndex(offset)] = static_cast<std::string::value_type>(
+      bits.to_ulong());  // should be safe since this really holds only 8 bits
+  return old_value;
+}
+
+// ------------------------------------------------------------------------- //
+// Helper functions to access the data or change it
+
+class OverRideValue {
+  long old_value_ = -1;
```

Comment:
I think the reason why we use data members is to preserve the state of a class. 
In `Set` function you assign `old_value_` and immediately return it. 
Sounds like something that  should be on stack and not  a data member